### PR TITLE
refactor(*): refactor premo to adapt to the new transaction structure

### PR DESCRIPTION
**Pull request by xiaochenl**

### CODE REVIEW

File: tester/bxh_tester/case004_xvm_test.go [hunk 1]
```diff
@@ -26,11 +28,14 @@ func (suite *Snake) TestDeployContractWithToAddress() {
 		Payload: contract,
 	}
 
+	payload, err := td.Marshal()
```
Comment (Alexader) [inline tester/bxh_tester/case004_xvm_test.go:31]: require nil for error

File: tester/bxh_tester/case005_readonly_test.go [hunk 2]
```diff
@@ -87,11 +87,14 @@ func genContractTransaction(
...
 	}
 
+	payload, err := td.Marshal()
+
 	tx := &pb.Transaction{
 		From:      from,
 		To:        address,
-		Data:      td,
 		Timestamp: time.Now().UnixNano(),
+		Nonce:     rand.Uint64(),
```
Comment (Alexader) [inline tester/bxh_tester/case005_readonly_test.go:96]: remove nonce

File: tester/bxh_tester/case005_readonly_test.go [hunk 3]
```diff
@@ -87,11 +87,14 @@ func genContractTransaction(
...
+		Payload:   payload,
 	}
 
 	if err := tx.Sign(privateKey); err != nil {
```
Comment (Alexader) [inline tester/bxh_tester/case005_readonly_test.go:100]: remove sign for tx

**Review by Alexader — CHANGES_REQUESTED** [review]

**Comment by xiaochenl** [comment]
The new pr has been resubmitted, its changes are more complete, and it runs more correctly

**Pull request closed without merging**